ethersjs-api-check/test/test.ts

```diff
@@ -0,0 +1,44 @@
+import {ethers} from 'ethers';
+import {assert} from 'chai';
+import 'mocha';
+
+describe('ethers.js compatibility', () => {
+
+	it('block retrieved with gasLimit', async () => {
+		let provider = new ethers.providers.JsonRpcProvider(process.env.npm_config_networkaddr);
+		let block = await provider.getBlock(process.env.npm_config_blocknum as string);
+
+		// These assertions trigger on undefined or null
+		assert.notEqual(block, null);
+		assert.notEqual(block.gasLimit, null);
+	});
+
+	it('EIP-1559 transactions supported', async () => {
+		let provider = new ethers.providers.JsonRpcProvider(process.env.npm_config_networkaddr);
+
+		// The fee data is the construct used to determine if EIP-1559 transactions are supported, if it contains max
+		let feeData = await provider.getFeeData();
+
+		// These assertions trigger on undefined or null
+		assert.notEqual(feeData, null);
+		// If the following 2 fields are set then the network is assumed to support EIP-1559 transactions.
+		assert.notEqual(feeData.maxFeePerGas, null);
+		assert.notEqual(feeData.maxPriorityFeePerGas, null);
+		// We check the other 2 fields for completeness, they should also be set.
+		assert.notEqual(feeData.gasPrice, null);
+		assert.notEqual(feeData.lastBaseFeePerGas, null);
+	});
+
+	// Our blockchain client implementation returns a default gas limit when the
+	// actual gas limit cannot be retrieved.  We cannot check fee data against a
+	// pruned block because getFeeData always requests the latest block.
+	it('block with pruned state still reports gas limit', async () => {
```

Comment:
Trying to understand something.
This test will be triggered probably with state.
But you are reusing this test in the e2e one, to execute with state, prune it, and execute it again right? So that is the actual test that is checking the prunning, and this one executed by circle, will always return what we have in the state

Reply:
For the pruned one, you could also add a test checking that the baseFee is null using the same logic (if we keep the baseFee as it is, and not adding a default as I asked in another comment)

Reply:
@ponti, I don't understand why you think this test would be executed with state?

Reply:
Ok so just to clarify, the e2e test was controlling which test to run from the set of tests and would only run the tests for the the pruned state when it had pruned the state.